pkg/sensors/base/base.go

```diff
@@ -164,24 +154,21 @@ func GetDefaultMaps(cgroupRate bool) []*program.Map {
 
 }
 
-// GetInitialSensor returns the base sensor
-func GetInitialSensor() *sensors.Sensor {
-	sensorInit.Do(func() {
-		setupPrograms()
-		sensor.Progs = GetDefaultPrograms(option.CgroupRateEnabled())
-		sensor.Maps = GetDefaultMaps(option.CgroupRateEnabled())
-	})
+func initBaseSensor() *sensors.Sensor {
+	sensor := sensors.Sensor{
+		Name: basePolicy,
+	}
+	setupPrograms()
+	sensor.Progs = GetDefaultPrograms(option.CgroupRateEnabled())
+	sensor.Maps = GetDefaultMaps(option.CgroupRateEnabled())
 	return &sensor
 }
 
-func GetInitialSensorTest() *sensors.Sensor {
-	sensorTestInit.Do(func() {
-		setupPrograms()
-		sensorTest.Progs = GetDefaultPrograms(true)
-		sensorTest.Maps = GetDefaultMaps(true)
```

Comment:
I took a look at this early as it's still in draft but here we had both options forced to `true` in the test version, is that okay that we now always pass the options' value?

Reply:
There are only two tests that use this, and they are setting the configuration variable. The last patch ensures that we are generating a new list of programs/maps for each test, which is the right thing to do I think.